firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/TesterSignInManagerTest.java

```diff
@@ -228,18 +236,37 @@ public void signInTester_whenSignInCalledMultipleTimes_secondCallHasNoEffect()
   public void signInTester_whenReturnFromSignIn_taskSucceeds()
       throws InterruptedException, FirebaseAppDistributionException, ExecutionException {
     Task signInTask = testerSignInManager.signInTester();
+    awaitAsyncOperations(backgroundExecutor);
+    awaitAsyncOperations(blockingExecutor);
 
     // Simulate re-entering app after successful sign in, via SignInResultActivity
     testerSignInManager.onActivityCreated(mockSignInResultActivity);
     awaitTask(signInTask);
 
     assertTrue(signInTask.isSuccessful());
-    verify(mockSignInStorage).setSignInStatus(true);
+    assertThat(signInStorage.getSignInStatusBlocking()).isTrue();
+  }
+
+  @Test
+  public void signInTester_whenStorageFailsToRecordSignInStatus_taskFails()
+      throws InterruptedException {
+    Exception expectedException = new RuntimeException("Error");
+    when(signInStorage.setSignInStatus(anyBoolean()))
+        .thenReturn(Tasks.forException(expectedException));
+    Task signInTask = testerSignInManager.signInTester();
+    awaitAsyncOperations(backgroundExecutor);
+    awaitAsyncOperations(blockingExecutor);
+
+    // Simulate re-entering app after successful sign in, via SignInResultActivity
+    testerSignInManager.onActivityCreated(mockSignInResultActivity);
+
+    awaitTaskFailure(signInTask, UNKNOWN, "Error storing tester sign in state", expectedException);
```

Comment:
nit: long line

Reply:
`:googleJavaFormat` says this is fine too. Looks like 100 chars is the max.